lib/vrl/stdlib/src/parse_regex_all.rs

```diff
@@ -78,6 +92,50 @@ impl Function for ParseRegexAll {
             },
         ]
     }
+
+    fn compile_argument(
+        &self,
+        _args: &[(&'static str, Option<FunctionArgument>)],
+        name: &str,
+        expr: Option<&expression::Expr>,
+    ) -> CompiledArgument {
+        match (name, expr) {
+            ("pattern", Some(expr)) => {
+                let regex: regex::Regex = match expr {
+                    expression::Expr::Literal(expression::Literal::Regex(regex)) => {
+                        Ok((**regex).clone())
```

Comment:
Don't Rust's auto-deref rules apply here to make it possible to call `regex.clone()`?

Reply:
The `regex` here is actually another wrapper over  `regex::Regex` https://github.com/vectordotdev/vector/blob/master/lib/vrl/compiler/src/value/regex.rs#L7

We need to deref to ensure we get the `regex::Regex` and not the wrapper.